aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AWSCognitoAuthPlugin.java

```diff
@@ -757,22 +738,15 @@ public void onResult(List<UserCodeDeliveryDetails> result) {
                                             Collections.emptyMap(),
                                             null)
                             ));
-                        }
-
-                        for (UserCodeDeliveryDetails details : result) {
-                            if (details.getAttributeName().equals(attribute.getKey().getKeyString())) {
+                        } else {
+                            for (UserCodeDeliveryDetails details : result) {
```

Comment:
What is this loop for?

Reply:
Since result is a list, we must go through the it to get the detail (even though in this case there is only one result)

Reply:
Since there's only one result you get it by calling `result.get(0)`

Reply:
Make sense. I've already changed that